src/components/common/DynamicListForm.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { EmptyState, EmptyStateBody } from "@patternfly/react-core/dist/esm/components/EmptyState";
+import { FormFieldGroup, FormFieldGroupHeader } from "@patternfly/react-core/dist/esm/components/Form";
+import { HelperText, HelperTextItem } from "@patternfly/react-core/dist/esm/components/HelperText";
+
+import './DynamicListForm.scss';
+
+export class DynamicListForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            list: [],
+        };
+        this.keyCounter = 0;
+        this.removeItem = this.removeItem.bind(this);
+        this.addItem = this.addItem.bind(this);
+        this.onItemChange = this.onItemChange.bind(this);
+    }
+
+    removeItem(idx, field, value) {
+        this.setState(state => {
+            const items = state.list.concat();
+            items.splice(idx, 1);
+            return { list: items };
+        }, () => this.props.onChange(this.state.list.concat()));
+    }
+
+    addItem() {
+        this.setState(state => {
+            return { list: [...state.list, Object.assign({ key: this.keyCounter++ }, this.props.default)] };
+        }, () => this.props.onChange(this.state.list.concat()));
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20230714-115437-e66ad82a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)